src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -407,8 +407,7 @@ private TypeParameterConstraintClause GetDefaultTypeParameterConstraintClause(Ty
                 return false;
             }
 
-            // Ignore nullability when comparing constraints.
-            if (constraintTypes.Contains(c => type.Equals(c, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes)))
+            if (constraintTypes.Contains(c => type.Equals(c, TypeCompareKind.AllIgnoreOptions & ~TypeCompareKind.ObliviousNullableModifierMatchesAny)))
```

Comment:
>~TypeCompareKind.ObliviousNullableModifierMatchesAny [](start = 96, length = 52)

Does this make any difference? Since we are ignoring any nullable differences.